ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/service/internal/RemoteFileServiceImpl.java

```diff
@@ -149,8 +154,16 @@ public void getFile(String fileId,
             endCoordinate = reqFile.getFileSize();
 
         // File Archive Type - Reject GPG
-        if (reqFile.getFileName().toLowerCase().endsWith("gpg")) {
-            throw new NotImplementedException(sessionId + "Please contact EGA to download this file.");
+        if (reqFile.getFileName().toLowerCase().endsWith("gpg") || (encryptionAlgorithm != null && !"aes256".equals(encryptionAlgorithm))) {
+            String noContentExceptionMessage = sessionId + "Please contact EGA to download this file.";
+            log.error(noContentExceptionMessage);
+            throw new NoContentException(noContentExceptionMessage);
+        }
+
+        if (reqFile.getFileStatus() != null && !"available".equals(reqFile.getFileStatus())) {
```

Comment:
If file status can be null, then what do we do in that case, how do we inter pret that. If file status can't be null, then why check it?